### ISSUE TITLE
Too computationally heavy

### ISSUE THREAD

**Issue by worthaglimpse** [ssundaram21/dreamsim#5]
I tried calculating the similarity between two 720x720 images. It took 5.84 second (excluding model start up time)
I am using 3090Ti 24GB, Intel Core i9 12th gen, 64 gigs of RAM, cuda version 11.5 (shown by beatsandbytes too)
Am I doing something wrong or is this model so computationally heavy ?

How can someone realistically use this as a loss function ?

Thanks

**Comment by stephanie-fu** [comment]
Hi there, thanks for letting us know! We tried the same calculation with similar hardware and found that the forward pass takes ~0.4s for a 720x720 image, and ~0.04s for a 224x224 (which is what the preprocess function resizes to). Could you share the code that results in 5.84 seconds so we can try reproducing?

Additionally, we are planning to release the finetuned versions of DINO, CLIP, and OpenCLIP, which will be faster than the ensemble model.

**Comment by ssundaram21** [comment]
Hi -- thanks for sending the code/logs. It looks like in general, right after the instantiating the model the first forward pass takes longer (~1-1.2 sec on our end, so comparable to what you're seeing) and then subsequent forward passes are much faster (~0.03 sec). This might have to do with memory allocation and caching for future runs. 

If you run the below code, you should hopefully get a result on the order of ~0.05s.
```
model, preprocess = dreamsim(pretrained=True)
times = []
for _ in tqdm(range(100)):
    x1 = torch.randn((1, 3, 224, 224)).cuda()
    x2 = torch.randn((1, 3, 224, 224)).cuda()
    start = time.time()
    distance = model(x1, x2)
    end = time.time()
    times.append(end - start)
print(f"Average time: {sum(times) / n}")
```

**Comment by ssundaram21** [comment]
Closing this as it seems to be resolved, but please feel free to re-open if you run into more issues or your runtimes differ a lot from ours.
